### PR TITLE
Unified docstring

### PR DESCRIPTION
**Pull request by PurityFan**
Reference of Tuners and Assessor:
Unified docstring of SMAC, Hyperband, MedianStop, Curvefitting

### CODE REVIEW

**Comment by leckie-chn** [comment]
Please refer to https://www.sphinx-doc.org/en/master/usage/extensions/example_numpy.html#example-numpy for docstring formatting.

**Comment by PurityFan** [comment]
> Please refer to https://www.sphinx-doc.org/en/master/usage/extensions/example_numpy.html#example-numpy for docstring formatting.

Done

File: src/sdk/pynni/nni/curvefitting_assessor/curvefitting_assessor.py [hunk 1]
```diff
@@ -51,9 +50,7 @@ def __init__(self, epoch_num=20, optimize_mode='maximize', start_step=6, thresho
         logger.info('Successfully initials the curvefitting assessor')
 
     def trial_end(self, trial_job_id, success):
-        '''
-        trial end: update the best performance of completed trial job
-        '''
+        """trial end: update the best performance of completed trial job"""
```
Comment (xuehui1991) [inline src/sdk/pynni/nni/curvefitting_assessor/curvefitting_assessor.py]: seems we need to point out the type of the parameters?

Comment (PurityFan) [reply src/sdk/pynni/nni/curvefitting_assessor/curvefitting_assessor.py]: Add already, please help review again :)

File: src/sdk/pynni/nni/curvefitting_assessor/curvefunctions.py [hunk 2]
```diff
@@ -25,7 +25,7 @@
                             'exp4', 'ilog2', 'weibull', 'janoschek']
 
 def vap(x, a, b, c):
-    ''' Vapor pressure model '''
+    """Vapor pressure model"""
```
Comment (xuehui1991) [inline src/sdk/pynni/nni/curvefitting_assessor/curvefunctions.py]: the type of a, b , c?

File: src/sdk/pynni/nni/curvefitting_assessor/curvefunctions.py [hunk 3]
```diff
@@ -77,10 +77,9 @@ def pow4(x, alpha, a, b, c):
 model_para_num['pow4'] = 4
 
 def mmf(x, alpha, beta, kappa, delta):
-    '''
-    Morgan-Mercer-Flodin
+    """Morgan-Mercer-Flodin
```
Comment (xuehui1991) [inline src/sdk/pynni/nni/curvefitting_assessor/curvefunctions.py]: the type of alpha and ...???

File: src/sdk/pynni/nni/curvefitting_assessor/curvefunctions.py [hunk 4]
```diff
@@ -102,18 +101,17 @@ def ilog2(x, c, a):
 model_para_num['ilog2'] = 2
 
 def weibull(x, alpha, beta, kappa, delta):
-    '''
-    Weibull model
+    """Weibull model
     http://www.pisces-conservation.com/growthhelp/index.html?morgan_mercer_floden.htm
-    '''
+    """
```
Comment (xuehui1991) [inline src/sdk/pynni/nni/curvefitting_assessor/curvefunctions.py]: same here

File: src/sdk/pynni/nni/curvefitting_assessor/curvefitting_assessor.py [hunk 5]
```diff
@@ -22,11 +22,21 @@
...
+    """CurvefittingAssessor uses learning curve fitting algorithm to predict the learning curve performance in the future.
     It stops a pending trial X at step S if the trial's forecast result at target step is convergence and lower than the
     best performance in the history.
-    '''
+
+    Parameters
+    ----------
+    epoch_num: int
+        The total number of epoch
+    optimize_mode: 'maximize' or 'minimize'
```
[the remark this answers is not in the document]
Comment (PurityFan) [reply src/sdk/pynni/nni/curvefitting_assessor/curvefitting_assessor.py]: Done. Other places have changed too.

File: src/sdk/pynni/nni/curvefitting_assessor/curvefitting_assessor.py [hunk 6]
```diff
@@ -51,9 +61,15 @@ def __init__(self, epoch_num=20, optimize_mode='maximize', start_step=6, thresho
...
-        '''
-        trial end: update the best performance of completed trial job
-        '''
+        """update the best performance of completed trial job
+        
+        Parameters
+        ----------
+        trial_job_id: int
+            trial job id
+        success: boolean
```
[the remark this answers is not in the document]
Comment (PurityFan) [reply src/sdk/pynni/nni/curvefitting_assessor/curvefitting_assessor.py]: Done. Other places have changed too.

File: src/sdk/pynni/nni/curvefitting_assessor/curvefunctions.py (hunk 2 above)
Comment (PurityFan) [reply src/sdk/pynni/nni/curvefitting_assessor/curvefunctions.py]: Done. Other places have changed too.

File: src/sdk/pynni/nni/curvefitting_assessor/curvefunctions.py (hunk 3 above)
Comment (PurityFan) [reply src/sdk/pynni/nni/curvefitting_assessor/curvefunctions.py]: Done. Other places have changed too.

File: src/sdk/pynni/nni/curvefitting_assessor/curvefunctions.py (hunk 4 above)
Comment (PurityFan) [reply src/sdk/pynni/nni/curvefitting_assessor/curvefunctions.py]: Done. Other places have changed too.

**Pull request merged**